bindings/node/src/pre_tokenizers.rs

```diff
@@ -0,0 +1,248 @@
+use tokenizers as tk;
+
+#[napi(string_enum)]
+pub enum JsSplitDelimiterBehavior {
+  Removed,
+  Isolated,
+  MergedWithPrevious,
+  MergedWithNext,
+  Contiguous,
+}
+
+impl TryFrom<String> for JsSplitDelimiterBehavior {
+  type Error = Error;
+
+  fn try_from(value: String) -> Result<Self> {
+    match &value[..] {
+      "removed" => Ok(JsSplitDelimiterBehavior::Removed),
+      "isolated" => Ok(JsSplitDelimiterBehavior::Isolated),
+      "mergedWithPrevious" => Ok(JsSplitDelimiterBehavior::MergedWithPrevious),
+      "mergedWithNext" => Ok(JsSplitDelimiterBehavior::MergedWithNext),
+      "contiguous" => Ok(JsSplitDelimiterBehavior::Contiguous),
+      _ => Err(Error::from_reason(
+        "Wrong value for SplitDelimiterBehavior, expected one of: \
+                 `removed, isolated, mergedWithPrevious, mergedWithNext, contiguous`"
+          .to_string(),
+      )),
+    }
+  }
+}
+
+impl From<JsSplitDelimiterBehavior> for SplitDelimiterBehavior {
+  fn from(value: JsSplitDelimiterBehavior) -> Self {
+    match value {
+      JsSplitDelimiterBehavior::Removed => SplitDelimiterBehavior::Removed,
+      JsSplitDelimiterBehavior::Isolated => SplitDelimiterBehavior::Isolated,
+      JsSplitDelimiterBehavior::MergedWithPrevious => SplitDelimiterBehavior::MergedWithPrevious,
+      JsSplitDelimiterBehavior::MergedWithNext => SplitDelimiterBehavior::MergedWithNext,
+      JsSplitDelimiterBehavior::Contiguous => SplitDelimiterBehavior::Contiguous,
+    }
+  }
+}
+
+/// PreTokenizers
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[napi]
+pub struct PreTokenizer {
+  #[serde(flatten, with = "arc_rwlock_serde")]
+  pretok: Option<Arc<RwLock<PreTokenizerWrapper>>>,
+}
+
+impl tk::PreTokenizer for PreTokenizer {
+  fn pre_tokenize(&self, pretokenized: &mut PreTokenizedString) -> tk::Result<()> {
+    self
+      .pretok
+      .as_ref()
+      .ok_or("Uninitialized PreTokenizer")?
+      .read()
+      .unwrap()
+      .pre_tokenize(pretokenized)?;
+
+    Ok(())
+  }
+}
+
+#[napi]
+impl PreTokenizer {
+  #[napi(ts_return_type = "[string, [number, number]][]")]
+  pub fn pre_tokenize_string(&self, sequence: String, env: Env) -> Result<Vec<Array>> {
+    use tk::PreTokenizer;
+
+    let mut pretokenized = PreTokenizedString::from(sequence);
+
+    self
+      .pre_tokenize(&mut pretokenized)
+      .map_err(|e| Error::from_reason(format!("{}", e)))?;
+
+    pretokenized
+      .get_splits(tk::OffsetReferential::Original, tk::OffsetType::Char)
+      .into_iter()
+      .map(|(s, (start, end), _)| -> Result<Array> {
+        let mut arr = env.create_array(2)?;
+        let mut offset = env.create_array(2)?;
+        offset.set(0, env.create_uint32(start as u32)?)?;
+        offset.set(1, env.create_uint32(end as u32)?)?;
+        arr.set(0, env.create_string(s)?)?;
+        arr.set(1, offset)?;
+        Ok(arr)
+      })
+      .collect::<Result<Vec<_>>>()
+  }
+}
+
+/// byte_level(addPrefixSpace: bool = true, useRegex: bool = true)
```

Comment:
How do you enforce `= true` here?

Reply:
The `unwrap_or`. 

All these comments existed before and are informative which is why I kept them here. They are doc comments not just comments.